src/main/java/eu/dissco/backend/repository/ElasticSearchRepository.java

```diff
@@ -187,13 +198,16 @@ private Pair<Long, List<DigitalSpecimenWrapper>> getDigitalSpecimenSearchResults
     }
   }
 
-  public Map<String, Map<String, Long>> getAggregations(Map<String, List<String>> params)
+  public Map<String, Map<String, Long>> getAggregations(Map<String, List<String>> params,
+      Set<MappingTerm> aggregationTerms, boolean isTaxonomyOnly)
       throws IOException {
     var aggregationQueries = new HashMap<String, Aggregation>();
     var queries = generateQueries(params);
-    for (var aggregationTerm : getAggregationList()) {
-      aggregationQueries.put(aggregationTerm.getName(), AggregationBuilders.terms()
-          .field(aggregationTerm.getFullName()).build()._toAggregation());
+    var size = isTaxonomyOnly ? Integer.MAX_VALUE : 10;
```

Comment:
What does size mean with aggregations? is 10 the default size normally?

Reply:
Yes default size is 10 but for taxonomy we would like to get back everything (as it is always filtered by level this shouldn't be thousands but might be hunderds). The frontend might then add some kind of limitation on it. Performance wise this is not an issue as it is just one filtered rank.

Reply:
For the regular aggregations (shown as the filters) 10 is fine.

Reply:
:+1: